src/code/actions.js

```diff
@@ -271,7 +272,9 @@ function _submitDrake(targetDrakeIndex, userDrakeIndex, correct, state) {
         userDrakeOrg = GeneticsUtils.convertDrakeToOrg(state.drakes[userDrakeIndex]),
         initialDrakeOrg = state.initialDrakes[userDrakeIndex] ? GeneticsUtils.convertDrakeToOrg(state.initialDrakes[userDrakeIndex]) : null,
         routeSpec = state.routeSpec,
-        editableGenes = state.authoring[routeSpec.level][routeSpec.mission][routeSpec.challenge].visibleGenes.split(", ");
+        challengeDefinition = AuthoringUtils.getChallengeDefinition(state.authoring, routeSpec),
+        editableGeneString = challengeDefinition.visibleGenes || challengeDefinition.userChangeableGenes,
```

Comment:
This originally came out of a request from Rob shortly before the classroom test. We need to circle back with him to figure out what the ITS is expecting and how it's interpreting these fields. We shouldn't be muddying the waters here. Also, @ekosmin , there will be a conflict here as I also addressed a crash that occurred in older challenges that didn't have the requisite fields. You should rebase to master before making further changes here.